catalogue/webapp/pages/images.tsx

```diff
@@ -115,6 +111,12 @@ const Images: NextPage<Props> = ({
     };
   }, []);
 
+  const { setLink } = useContext(SearchContext);
```

Comment:
I might be inclined to extract this into a little `useSearchLink` hook

```typescript
const useSearchLink = (createLink, source, routeProps) => {
	const { setLink } = useContext(SearchContext);
	
	useEffect(() => {
		const link = createLink(routeProps, source);
		setLink(link);
	}, [routeProps]);
}

// ...


useSearchLink(imagesLink, imagesRouteProps, "works_search_context")
```

or something like that

Reply:
Nice suggestion - I'll have a play when merging this into #6101